base/show.jl

```diff
@@ -993,10 +993,10 @@ function show_datatype(io::IO, @nospecialize(x::DataType), wheres::Vector=TypeVa
     # Print homogeneous tuples with more than 3 elements compactly as NTuple{N, T}
     if istuple
         if n > 3 && all(@nospecialize(i) -> (parameters[1] === i), parameters)
-            print(io, "NTuple{", n, ", ", parameters[1], "}")
+            print(io, "NTuple{", n, ", ", repr(parameters[1]), "}")
```

Comment:
This should call `show(io, parameters[1])` (same below) so that the context information in `io` can be used.